frontends/p4/typeChecking/typeCheckTypes.cpp

```diff
@@ -173,6 +173,8 @@ const IR::Node *TypeInferenceBase::postorder(const IR::Type_Specialized *type) {
         if (!argtype) return type;
         if (auto self = ContainsType::find(argtype, baseType, typeMap)) {
             typeError("%1%: contains self '%2%' as type argument", type->baseType, self);
+            // If we continue we could get a IR loop which would crash the compiler, so stop here.
+            FATAL_ERROR("Compilation cannot continue due to type errors");
             return type;
```

Comment:
just wondering if returning `nullptr` here would be better?

Reply:
Wouldn't that possibly trigger an error in `IR::Node::validate()` of the containing node?

Reply:
Do not know :) Things are a bit convoluted. Or maybe `prune()` in case of errors is to be used so we won't process child nodes of potentially invalid types, etc.